pkg/watches/watches.go

```diff
@@ -33,13 +33,13 @@ import (
 
 type Watch struct {
 	schema.GroupVersionKind `json:",inline"`
-	ChartPath               string `json:"chart"`
+	// Though the name is ChartDir, this accepts chart directory or a chart path.
+	ChartDir string `json:"chart"`
```

Comment:
ChartPath - accepted both chart directory and chart tar.gz
ChartDir - accepted only the directory path

The reason I have just changed the name and not modified the loader, is that though we accept both inputs internally the user will not see any change. When we are ready to make a breaking change in helm, we can revert this back `ChartPath`. Is it fine?

Reply:
Would not make life easier to ensure that is == helm plugin in SDK and we plan any rename for another plugin version where we ensure the same changes in both? 
How hard will be for us and for the users to remember the diffs and etc?

Reply:
The reason for not versioning this as a separately is we want hybrid plugin scaffolding without any modifications to be exactly same as the helm plugin which we have now in SDK. This is to maintain backwards compatibility and enable helm authors to move to hybrid with out making any breaking changes.

Reply:
There's two pieces of this API right? There's the Go `Watch` struct and the `watches.yaml` file.

Correct me if I'm wrong:
- we want to avoid breaking the `watches.yaml` format, so we need to keep the json field name set as `chart`
- we don't care about the Go `Watch` struct because it has been an internal API in the `operator-sdk` repo.

If I'm understanding correctly, I think that means we can keep `ChartPath` as the name in the Go struct, especially since it would be confusing to use `ChartDir` and still accept tgz charts.

Reply:
@joelanford agreed! Had overlooked that json tag for both are same!